### ISSUE TITLE
dsconf fails to add a PTA URL due to an invalid check.

### ISSUE THREAD

**Issue by mreynolds389** [389ds/389-ds-base#4719]
When trying to configure the PTA plugin by adding an URL, the dsconf command fails.

Version-Release number of selected component (if applicable):
# cat /etc/redhat-release
Red Hat Enterprise Linux release 8.3 (Ootpa)
#
# rpm -qa | grep 389-ds-base-1
389-ds-base-1.4.3.13-1.module+el8dsrv+8334+69a46a2e.x86_64
#

How reproducible:
Always.

Steps to Reproduce:
Launch the command in verbose mode. It fails since it considers the DN as invalid:

# dsconf --verbose -D "cn=Directory Manager" ldap://localhost:10389 plugin pass-through-auth url add "ldap://localhost:7389/o=redhat"

...

DEBUG: Subtree is an invalid DN
Traceback (most recent call last):
  File "/usr/sbin/dsconf", line 134, in <module>
    result = args.func(inst, None, log, args)
  File "/usr/lib/python3.6/site-packages/lib389/cli_conf/plugins/passthroughauth.py", line 88, in pta_add
    new_url_l = _validate_url(args.URL.lower())
  File "/usr/lib/python3.6/site-packages/lib389/cli_conf/plugins/passthroughauth.py", line 58, in _validate_url
    raise ValueError("Subtree is an invalid DN")
ValueError: Subtree is an invalid DN
ERROR: Error: Subtree is an invalid DN

https://bugzilla.redhat.com/show_bug.cgi?id=1947469

**Comment by QUASARFREAK** [comment]
I'm having this exact same issue:

# cat /etc/redhat-release
Red Hat Enterprise Linux release 8.1 (Ootpa)

# rpm -qa | grep 389-ds-base-1
389-ds-base-1.4.3.21-3.module+el8dsrv+10401+3d549418.x86_64

# dsconf --verbose -D "cn=Directory Manager" ldap://localhost:10389 plugin pass-through-auth url add ldap://ldaporig:10389/ou=People,c=arg,O=BBVA
DEBUG: The 389 Directory Server Configuration Tool
DEBUG: Inspired by works of: ITS, The University of Adelaide
DEBUG: dsrc path: /root/.dsrc
DEBUG: dsrc container path: /data/config/container.inf
DEBUG: dsrc instances: []
DEBUG: dsrc no such section: slapd-ldap://localhost:10389
DEBUG: Called with: Namespace(URL='ldap://ldaporig:10389/ou=People,c=arg,O=BBVA', basedn=None, binddn='cn=Directory Manager', bindpw=None, func=<function pta_add at 0x7f68e927e8c8>, instance='ldap://localhost:10389', json=False, prompt=>
DEBUG: Instance details: {'uri': 'ldap://localhost:10389', 'basedn': None, 'binddn': 'cn=Directory Manager', 'bindpw': None, 'saslmech': None, 'tls_cacertdir': None, 'tls_cert': None, 'tls_key': None, 'tls_reqcert': None, 'starttls': Fa>
DEBUG: SER_SERVERID_PROP not provided, assuming non-local instance
DEBUG: Allocate <class 'lib389.DirSrv'> with ldap://localhost:10389
DEBUG: Allocate <class 'lib389.DirSrv'> with localhost:389
DEBUG: Allocate <class 'lib389.DirSrv'> with localhost:389
Enter password for cn=Directory Manager on ldap://localhost:10389:
DEBUG: SER_SERVERID_PROP not provided, assuming non-local instance
DEBUG: Allocate <class 'lib389.DirSrv'> with ldap://localhost:10389
DEBUG: Allocate <class 'lib389.DirSrv'> with localhost:389
DEBUG: Allocate <class 'lib389.DirSrv'> with localhost:389
DEBUG: open(): Connecting to uri ldap://localhost:10389
DEBUG: Using dirsrv ca certificate /etc/dirsrv/slapd-{instance_name}
DEBUG: Using external ca certificate /etc/dirsrv/slapd-{instance_name}
DEBUG: Using external ca certificate /etc/dirsrv/slapd-{instance_name}
DEBUG: Using /etc/openldap/ldap.conf certificate policy
DEBUG: ldap.OPT_X_TLS_REQUIRE_CERT = 2
DEBUG: open(): bound as cn=Directory Manager
DEBUG: Retrieving entry with [('',)]
DEBUG: Retrieved entry [dn:
vendorVersion: 389-Directory/1.4.3.21 B2021.077.1932

]
DEBUG: Subtree is an invalid DN
Traceback (most recent call last):
  File "/usr/sbin/dsconf", line 134, in <module>
    result = args.func(inst, None, log, args)
  File "/usr/lib/python3.6/site-packages/lib389/cli_conf/plugins/passthroughauth.py", line 88, in pta_add
    new_url_l = _validate_url(args.URL.lower())
  File "/usr/lib/python3.6/site-packages/lib389/cli_conf/plugins/passthroughauth.py", line 58, in _validate_url
    raise ValueError("Subtree is an invalid DN")
ValueError: Subtree is an invalid DN
ERROR: Error: Subtree is an invalid DN

**Comment by QUASARFREAK** [comment]
If I comment out the raise error there, the url adds without any issue and the PTA works like a expected

**Comment by mreynolds389** [comment]
Just fixed it this PR (not merged yet):  https://github.com/389ds/389-ds-base/pull/4774

```
diff --git a/src/lib389/lib389/cli_conf/plugins/passthroughauth.py b/src/lib389/lib389/cli_conf/plugins/passthroughauth.py
index 6f8e88c19..fd8264298 100644
--- a/src/lib389/lib389/cli_conf/plugins/passthroughauth.py
+++ b/src/lib389/lib389/cli_conf/plugins/passthroughauth.py
@@ -54,7 +54,7 @@ def _validate_url(url):
     if (":" not in link) or ("//" not in link) or ("/" not in link) or (params and "," not in params):
         failed = False
 
-    if ldap.dn.is_dn(link.split("/")[-1]):
+    if not ldap.dn.is_dn(link.split("/")[-1]):
         raise ValueError("Subtree is an invalid DN")
 
     if params and len(params.split(",")) != 6 and not all(map(str.isdigit, params.split(","))):
```

We'll get this into the next release of 389-ds-base-1.4.3

**Comment by mreynolds389** [comment]
Fixed in master via #4774 

7b5834702..23c28c2f0  389-ds-base-1.4.4 -> 389-ds-base-1.4.4
8537cee0d..8aa5ef7dd  389-ds-base-1.4.3 -> 389-ds-base-1.4.3
